### PR TITLE
bump up the troubleshoot version to v0.72.x

### PR DESCRIPTION
**Pull request by arcolife**
#### What this PR does / why we need it:
bump up the troubleshoot version to v0.71.1

#### Which issue(s) this PR fixes:
Fixes [SC-87575](https://app.shortcut.com/replicated/story/87575/upgrade-troubleshoot-to-v0-71-1)

#### Special notes for your reviewer:

- Follow-up from #4029 
- Turns out, for tests to kick in, a PR should be raised from a feature branch in upstream rather than the fork. 
  (🙇🏼‍♂️ @cbodonnell for the hint!). 
- Visual smoke tests on okteto went well (uploading a license to admin portal and running analysers).

#### Does this PR introduce a user-facing change?

NONE

#### Does this PR require documentation?
NONE

### CODE REVIEW

**Review by adamancini — APPROVED** [review]

**Comment by xavpaice** [comment]
Currently blocked by https://github.com/replicatedhq/troubleshoot/issues/1331, we'll need to integrate the version that has the fix

File: go.mod [hunk 1]
```diff
@@ -49,7 +49,7 @@ require (
 	github.com/pmezard/go-difflib v1.0.0
 	github.com/replicatedhq/kotskinds v0.0.0-20230724164735-f83482cc9cfe
 	github.com/replicatedhq/kurlkinds v1.3.6
-	github.com/replicatedhq/troubleshoot v0.70.3
+	github.com/replicatedhq/troubleshoot v0.71.1
```
Comment (xavpaice) [inline go.mod:52]: v0.72.1 has the fixes we need

**Comment by arcolife** [comment]
- `v0.72.1` released, but uses a new version of `github.com/Azure/azure-sdk-for-go/sdk/azcore` in https://github.com/replicatedhq/troubleshoot/pull/1335.
- OTOH, KOTS is using version `v0.21.1` due to the [replace directive](https://github.com/replicatedhq/kots/blob/main/go.mod#L416) in its go.mod file, which cancels the indirect requirement version.
- This leads to an error in `go mod tidy` after bumping troubleshoot itself.

```
go: finding module for package github.com/Azure/azure-sdk-for-go/sdk/azcore/cloud
go: github.com/replicatedhq/kots/pkg/kotsutil imports
        github.com/replicatedhq/troubleshoot/pkg/collect imports
        github.com/microsoft/go-mssqldb tested by
        github.com/microsoft/go-mssqldb.test imports
        github.com/microsoft/go-mssqldb/aecmk/akv imports
        github.com/Azure/azure-sdk-for-go/sdk/azidentity imports
        github.com/Azure/azure-sdk-for-go/sdk/azcore/cloud: module github.com/Azure/azure-sdk-for-go/sdk/azcore@latest found (v1.7.2, replaced by github.com/Azure/azure-sdk-for-go/sdk/azcore@v0.21.1), but does not contain package github.com/Azure/azure-sdk-for-go/sdk/azcore/cloud
```

File: go.mod (hunk 1 above)
Comment (arcolife) [reply go.mod:52]: tried but see my comment below

**Comment by cbodonnell** [comment]
@arcolife The key issue was the bump of `github.com/microsoft/go-mssqldb` to `v1.6.0` brought in by the latest troubleshoot go module.  `github.com/microsoft/go-mssqldb` depends on `github.com/Azure/azure-sdk-for-go/sdk/azcore/cloud`, which does not exist in `v0.21.1`.  Other than this, we are OK and since KOTS does not depend on troubleshoot functions using `github.com/microsoft/go-mssqldb`, we can pin `github.com/microsoft/go-mssqldb` at the existing `v1.5.0` version and it will build:

```
github.com/microsoft/go-mssqldb => github.com/microsoft/go-mssqldb v1.5.0
```

For some context, we have pinned these Azure dependencies to older versions because we also import Velero to support configuring snapshots in KOTS. Velero's `go.mod`: https://github.com/vmware-tanzu/velero/blob/main/go.mod#L65-L67

There are likely some `replace` directives outside of this that we could get rid of, but I agree with you that it's out of scope of this PR.

**Comment by arcolife** [comment]
> since KOTS does not depend on troubleshoot functions using github.com/microsoft/go-mssqldb

@cbodonnell Thanks for the confirmation. I've pinned it with a comment and if the tests pass (eventually), we're good to go!

File: go.mod [hunk 2]
```diff
@@ -83,9 +83,9 @@ require (
 	k8s.io/kubelet v0.23.6
 	k8s.io/utils v0.0.0-20230505201702-9f6742963106
 	sigs.k8s.io/application v0.8.3
-	sigs.k8s.io/controller-runtime v0.15.0
-	sigs.k8s.io/kustomize/api v0.14.0
-	sigs.k8s.io/kustomize/kyaml v0.14.3
+	sigs.k8s.io/controller-runtime v0.16.1
+	sigs.k8s.io/kustomize/api v0.13.5-0.20230601165947-6ce0bf390ce3
+	sigs.k8s.io/kustomize/kyaml v0.14.3-0.20230601165947-6ce0bf390ce3
```
[the remark this answers is not in the document]
Comment (arcolife) [reply go.mod]: yep, rebased. Didn't see a warning "out of date with base branch" here. 🤔  
it seems those were bumped in https://github.com/replicatedhq/kots/commit/28eceebcfb4b8a24a5117e0da4cc1a689d2fb916 which fell in between this work. But good catch!

[the remark this answers is not in the document]
Comment (cbodonnell) [reply go.mod]: I was surprised not to see that either since it's usually there. Strange...

**Review by cbodonnell — APPROVED** [review]

**Pull request merged**